### PR TITLE
ui/admin: convert file to typescript

### PR DESCRIPTION
**Pull request by tony-tvu**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
Part of an ongoing effort to convert existing javascript files to typescript.

**Which issue(s) this PR fixes:**
https://github.com/target/goalert/issues/2318

### CODE REVIEW

File: web/src/app/admin/AdminDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+interface FieldValues {
+  [id: string]: string
+}
+
+interface AdminDialogProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  mutation: DocumentNode | TypedDocumentNode<any, OperationVariables>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  values: Value[] | any
```
Comment (Forfold) [inline web/src/app/admin/AdminDialog.tsx:32]: Looks like the parent component is using `ConfigValue[]` as the type for this prop

```suggestion
  values: ConfigValue[]
```

File: web/src/app/admin/AdminDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+function AdminDialog(props: AdminDialogProps): JSX.Element {
+  const [commit, { error }] = useMutation(props.mutation, {
+    onCompleted: props.onComplete,
+  })
+  const changeKeys = Object.keys(props.fieldValues)
+  const changes = props.values
+    .filter((v: { id: string }) => changeKeys.includes(v.id))
+    .map((orig: { id: string | number; value: number; type: string }) => ({
```
Comment (Forfold) [inline web/src/app/admin/AdminDialog.tsx:45]: We can remove the type here once we set the above to `ConfigValue[]`

```suggestion
    .map((orig) => ({
```

File: web/src/app/admin/AdminDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  type: string
+}
+
+interface FieldValues {
+  [id: string]: string
+}
+
+interface AdminDialogProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  mutation: DocumentNode | TypedDocumentNode<any, OperationVariables>
```
Comment (Forfold) [inline web/src/app/admin/AdminDialog.tsx:30]: My IDE seems fine with using just `DocumentNode`, so we can remove the `any` reference

```suggestion
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  mutation: DocumentNode
```

File: web/src/app/admin/AdminDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+          },
+        })
+      }
+      primaryActionLabel='Confirm'
+      errors={errs}
+      form={
+        <List data-cy='confirmation-diff'>
+          {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            changes.map((c: any) => (
```
Comment (Forfold) [inline web/src/app/admin/AdminDialog.tsx:80]: Looks like this is inferred

```suggestion
            changes.map((c) => (
```

**Review by Forfold — CHANGES_REQUESTED** [review]

**Review by Forfold — APPROVED** [review]

**Review by mastercactapus — APPROVED** [review]

**Pull request merged**